### PR TITLE
Template functions in manifest templates

### PR DESCRIPTION
**Pull request by macat**
resolves https://github.com/namely/broadway/issues/82

### CODE REVIEW

File: deployment/manifest_test.go [hunk 1]
```diff
@@ -7,10 +7,30 @@ import (
...
+		name     string
+		template string
+		vars     map[string]string
+		expected string
+	}{
+		{
+			name:     "simple",
+			template: `{{ .test }}`,
+			vars:     map[string]string{"test": "hello!"},
+			expected: "hello!",
```
Comment (poopoothegorilla) [inline deployment/manifest_test.go]: It is alittle difficult to tell the different test cases without a `scenario` field or something similar.. is the `name` field supposed to be this as well?

**Comment by poopoothegorilla** [comment]
+1

**Comment by therealplato** [comment]
@macat can you add a test for `datetime`? it returns a Time object and i'm curious how that casts to a string for templating

File: deployment/manifest_test.go (hunk 1 above)
Comment (macat) [inline deployment/manifest_test.go]: that's how we named cases in other files

Comment (macat) [inline deployment/manifest_test.go]: Sorry, I'm wrong. I'll change it.

Comment (macat) [inline deployment/manifest_test.go]: We have examples for both. I'm fine with scenario

**Comment by macat** [comment]
It implements Stringer interface. So it'll return this. https://golang.org/src/time/format.go?s=12854:12883#L407

**Pull request merged**